src/marqo/tensor_search/validation.py

```diff
@@ -255,6 +256,50 @@ def validate_field_name(field_name) -> str:
         raise InvalidFieldNameError(f"field name can't be a protected field. Please rename this field: {field_name}")
 
 
+def validate_unstructured_index_field_name(field_name) -> str:
+    """TODO:
+        - length (remember the vector name will have the vector_prefix added to the front of field_name)
+        - consider blanket "no double names starting with double underscore..."
+    Args:
+        field_name:
+
+    returns field_name, if all validations pass
+
+    Raises:
+        InvalidFieldNameError
+    """
+
+    # TODO-Li Do we still have these restrictions for an unstructured index?
+    if not field_name:
```

Comment:
Unstructured name validation should be consistent with structured name validation (plus the extra restriction of `::`). See `validate_structured_field` in `marqo_index`. I suggest extracting field name validation into a `validate_field_name` function then calling that for both unstructured and structured + plus extra validation of `::` for unstructured. As noted below, do all of this in `UnstructuredVespaIndex` instead of here